woocommerce/woocommerce
# Add Checkbox support for Additional Fields in Checkout Block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/42134



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Use the following code for testing 

```php
add_action(
	'woocommerce_blocks_loaded',
	function() {
		woocommerce_blocks_register_checkout_field(
			array(
				'id'       => 'plugin-namespace/leave-at-porsh',
				'label'    => __( 'leave at porsh', 'woo-gutenberg-products-block' ),
				'location' => 'address',
				'type'     => 'checkbox',
			)
		);
		woocommerce_blocks_register_checkout_field(
			array(
				'id'             => 'plugin-namespace/vat-number',
				'label'          => __( 'VAT Number', 'woo-gutenberg-products-block' ),
				'optionalLabel'  => __( 'VAT Number (optional)', 'woo-gutenberg-products-block' ),
				'autocomplete'   => 'vat-number',
				'autocapitalize' => 'characters',
				'location'       => 'address',
				'type'           => 'text',
			)
		);
	}
);
```
1. Add an item to your cart and go to the Checkout block.
2. In the address block, ensure you have a "Leave with Neighbor" checkbox and a vat text field.
3. Make sure it's on its own line.
4. Check the checkbox.
5. Reload the page, make sure the checkbox is still checked.
6. Uncheck the checkbox, make sure you can place the order.

#### Store API tests

<details><summary>Request body</summary>
<p>
```json
{
  "shipping_address": {
    "first_name": "Jon",
    "last_name": "Doe",
    "company": "",
    "address_1": "30 Test road",
    "address_2": "",
    "city": "Testville",
    "state": "CA",
    "postcode": "90210",
    "country": "US",
    "phone": ""
  },
  "billing_address": {
    "first_name": "Jon",
    "last_name": "Doe",
    "company": "",
    "address_1": "30 Test road",
    "address_2": "",
    "city": "Testville",
    "state": "CA",
    "postcode": "90210",
    "country": "US",
    "phone": "",
    "email": "jon@doe.com",
    "phone": ""
  },
  "customer_note": "",
  "create_account": false,
  "payment_method": "bacs",
  "payment_data": [
    {
      "key": "wc-bacs-new-payment-method",
      "value": false
    }
  ]
}
```
</p>
</details> 

 1. Make sure you place an order without passing in the checkbox value.
 2. Make sure you can't place an order with a string for value (`"1"` and `"0"` are still parsed as boolean).


### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
